### PR TITLE
Fix paths containg src dir

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/path_rewriting.rs [hunk 1]
```diff
@@ -73,14 +90,14 @@ fn fixup_rel_path(source_dir: &Option<PathBuf>, abs_path: &PathBuf, rel_path: Pa
 }
 
 // Get the absolute path for the source file's path, resolving symlinks.
-fn get_abs_path(source_dir: &Option<PathBuf>, rel_path: PathBuf) -> (PathBuf, PathBuf) {
+fn get_abs_path(source_dir: &Option<PathBuf>, rel_path: PathBuf, cache: &mut Option<PathBuf>) -> (PathBuf, PathBuf) {
     let mut abs_path = if !rel_path.is_relative() {
         rel_path.clone()
     } else if let Some(ref source_dir) = source_dir {
         if !cfg!(windows) {
```
Comment (marco-c) [inline src/path_rewriting.rs]: Nit: you could do:
```
let rel_path = if !cfg!(windows) {
    rel_path
} else {
    PathBuf::from(&rel_path.to_str().unwrap().replace("/", "\\"))
}
guess_abs_path(&source_dir, &rel_path, cache)
```

Comment (calixteman) [reply src/path_rewriting.rs]: If I do that, then it doesn't compile since rel_path is moved and so no more usable in fixup call

Comment (marco-c) [reply src/path_rewriting.rs]: What if instead you do:
```Rust
let rel_path = if !cfg!(windows) {
    &rel_path
} else {
    &PathBuf::from(&rel_path.to_str().unwrap().replace("/", "\\"))
}
guess_abs_path(&source_dir, rel_path, cache)
```

(you could also change the name of `rel_path` if that makes it more readable)

Comment (calixteman) [reply src/path_rewriting.rs]: PathBuf::from.... doesn't live enough to get a ref on it

**Pull request merged**